floodlight/core/pitch.py

```diff
@@ -147,3 +153,967 @@ def center(self):
+        ax = ax or plt.subplots(**kwargs)[1]
+        ax.set_aspect(1)  # ensures that the ratio between length and width is correct
+        # regardless of the figsize
+
+        # create axes with handball pitch
+        if sport == "handball":
+            return self._plot_handball_pitch(
+                scalex=scalex,
+                scaley=scaley,
+                color_scheme=color_scheme,
+                save=save,
+                ax=ax,
+                **kwargs,
+            )
+
+        # create axes with soccer/football pitch
+        if sport == "soccer" or "football":
+            return self._plot_soccer_pitch(
+                scalex=scalex,
+                scaley=scaley,
+                color_scheme=color_scheme,
+                save=save,
+                ax=ax,
+                **kwargs,
+            )
+
+    def _plot_handball_pitch(
+        self,
+        scalex: bool,
+        scaley: bool,
+        color_scheme: str,
+        save: bool,
+        ax: plt.axes,
+        **kwargs,
+    ):
+
+        """Plots a handball pitch on a given axes
+
+        Parameters
+        ----------
+        scalex: bool
+            Determines if the view limits are adapted to the
+            data limits of the x values.
+        scaley: bool
+            Determines if the view limits are adapted to the
+            data limits of the y values.
+        color_scheme: str
+             Color scheme of the plot. One of {'normal', 'black_white'}.
+             If not given 'normal' is the default color scheme.
+        save: bool
+            You may want to save the graph. If the graphic should be saved 'save'
+            must be True. If not given as an argument the pitch is not going to be
+            saved. Plots are going to be stored as .png on the same directory layer as
+            the repository.
+        kwargs:
+            You may pass optional arguments (`linewidth`} which can be used for the plot
+            with matplotlib.
+        ax: plt.axes
+            Axes from matplotlib library on which the handball field is plotted.
+
+        Returns
+        ----------
+        ax : matplotlib.axes._subplots.AxesSubplot
+            An axes to which all elements of the sport specific pitch are added.
+        """
+
+        # customizing visualization
+        if color_scheme == "black_white":
+            ax.set_facecolor("white")
+            outline = "black"
+            innerline = "black"
+            fill = "white"
+            goal = "black"
+        else:
+            ax.set_facecolor("skyblue")
+            outline = "white"
+            innerline = "white"
+            fill = "khaki"
+            goal = "white"
+
+        # key data for the handball pitch
+        xmin, xmax = self.xlim
+        ymin, ymax = self.ylim
+
+        x_range = xmax - xmin
+        y_range = ymax - ymin
+        x_half = (xmax + xmin) / 2
+        y_half = (ymax + ymin) / 2
+
+        linewidth = kwargs.get("linewidth", 1)
```

Comment:
it seems to me that linewidth is the only kwarg you are handling, what's with the others? couldn't you just pass the entire **kwargs dict to the respective plot functions and let matplotlib do the rest?

Reply:
Yes and no, it really depends on which arguments we get from the pitch.plot function call.
I think the smartest way would be to just pass the kwargs down to the matplotlib.plot function. But that means that we have to either (1) filter all kwargs that are not accepted by the matplotlib.plot function or (2) add all arguments that can't be passed to matplotlib.plot functions to the plot functions arguments. Imo the (2) is more advantageous.

With (2) the plot function would look like this:

plot(self, sport: str = None, [other arguments in the same style like color_scheme, save, figsize, ax], **kwargs) 
the **kwargs must be identifiable by matplotlib plot functions.

I would like to hear your opinion on how to handle this.